pkg/network/ebpf/c/runtime/conntrack.c

```diff
@@ -28,37 +29,43 @@ int kprobe___nf_conntrack_hash_insert(struct pt_regs* ctx) {
         increment_telemetry_count(registers_dropped);
         return 0;
     }
+    
+    log_debug("kprobe/__nf_conntrack_hash_insert: netns: %u, status: %x\n", get_netns(&ct->ct_net), status);
+
+    conntrack_tuple_t orig = {}, reply = {};
+    if (nf_conn_to_conntrack_tuples(ct, &orig, &reply) != 0) {
+        return 0;
+    }
 
-    struct nf_conntrack_tuple_hash tuplehash[IP_CT_DIR_MAX];
-    __builtin_memset(tuplehash, 0, sizeof(tuplehash));
-    bpf_probe_read(&tuplehash, sizeof(tuplehash), &ct->tuplehash);
+    bpf_map_update_elem(&conntrack, &orig, &reply, BPF_ANY);
+    bpf_map_update_elem(&conntrack, &reply, &orig, BPF_ANY);
+    increment_telemetry_count(registers);
 
-    struct nf_conntrack_tuple orig = tuplehash[IP_CT_DIR_ORIGINAL].tuple;
-    struct nf_conntrack_tuple reply = tuplehash[IP_CT_DIR_REPLY].tuple;
+    return 0;
+}
 
-    u32 netns = get_netns(&ct->ct_net);
-    log_debug("kprobe/__nf_conntrack_hash_insert: netns: %u, status: %x\n", netns, status);
+SEC("kprobe/ctnetlink_fill_info")
+int kprobe_ctnetlink_fill_info(struct pt_regs* ctx) {
```

Comment:
Something that just occurred to me. Do we need to filter to only the system-probe process, so that we aren't reacting to netlink dumps from other process/tools like `conntrack`?